tests/backends/test_pyevm.py

```diff
@@ -25,4 +44,23 @@ def eth_tester():
 
 
 class TestPyEVMBackendDirect(BaseTestBackendDirect):
-    pass
+    def test_can_estimate_gas_after_exception_raised_estimating_gas(self, eth_tester):
+        self.skip_if_no_evm_execution()
+
+        throws_address = _deploy_throws(eth_tester)
+        call_will_throw_transaction = _make_call_throws_transaction(
+            eth_tester,
+            throws_address,
+            'willThrow',
+        )
+        with pytest.raises(InvalidInstruction):
+            eth_tester.estimate_gas(dissoc(call_will_throw_transaction, 'gas'))
```

Comment:
We shouldn't be exposing internal py-evm exceptions through eth-tester. The exceptions raised should be the same no matter the back end. Looks like we need to define a new exception, then convert it. Here is an example: https://github.com/ethereum/eth-tester/blob/master/eth_tester/backends/pyevm/main.py#L331

Reply:
Should py-evm change to indicate an error has occurred with the `estimate_gas`, without exposing the exception?

Reply:
Not exactly sure what you're saying. I think we should add a new `eth_tester.exceptions.InvalidInstruction` (or something like it), and in the py-evm backend for `estimate_gas()`, we should capture the py-evm `InvalidInstruction` and raise the eth_tester one.

Reply:
I get it now.  I rebased and broke that line reference.  I added InvalidInstruction and Revert (:balloon: :baloon: Byzantium :balloon::balloon:)